library/src/test/java/com/getstream/sdk/chat/viewmodel/CreateChannelViewModelTest.kt

```diff
@@ -42,43 +54,52 @@ internal class CreateChannelViewModelTest {
         whenever(createChannelResult.data()) doReturn CHANNEL
         whenever(createChannelResult.isError) doReturn false
         whenever(createChannelResult.isSuccess) doReturn true
+
+        Dispatchers.setMain(testCoroutineDispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+        testCoroutineScope.cleanupTestCoroutines()
     }
 
     @Test
     fun `Should inform about validation error`() {
         whenever(createChannelResult.isError) doReturn false
         whenever(createChannelResult.isSuccess) doReturn true
-        val viewModel = CreateChannelViewModel(domain = chatDomain, client = chatClient)
-        val states = viewModel.state.observeAll()
-        val channelNameCandidate = ""
+        val channelNameCandidates =
```

Comment:
Can't we use `ParameterizedTest`?
If we use it instead of a list and anyone fails, all cases are run and show which ones are failing, using a list and iterating over them into the test method, it will fail on the first one but not check the rest of them